### PR TITLE
Use presenters for the show actions

### LINKED ISSUE #159: Too many LDP requests on show page
```
Started GET "/concern/documents/g732d8963" for ::1 at 2015-08-18 16:10:55 -0500
Processing by CurationConcerns::DocumentsController#show as HTML
  Parameters: {"id"=>"g732d8963"}
```
loading the work (Image):
```
  Load LDP (158.7ms) http://127.0.0.1:8983/fedora/rest/dev/g7/32/d8/96/g732d8963 Service: 70200474251280
  User Load (0.3ms)  SELECT  "users".* FROM "users" WHERE "users"."id" = ?  ORDER BY "users"."id" ASC LIMIT 1  [["id", 1]]
Usergroups are ["public", "registered"]
[CANCAN] Checking read permissions for user: brian@example.com with groups: ["public", "registered"]
[CANCAN] edit_groups: []
[CANCAN] read_groups: ["registered"]
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/curation_concerns/generic_files/_media_display.html.erb (20.5ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/curation_concerns/base/_representative_media.html.erb (54.3ms)
```
loading permission on the Image: read, registered
```
  Load LDP (151.4ms) http://127.0.0.1:8983/fedora/rest/dev/63/f4/5f/05/63f45f05-df91-4c78-a38b-d3288229f507 Service: 70200474251280
```
loading permission on the Image: write, brian@example.com
```
  Load LDP (142.3ms) http://127.0.0.1:8983/fedora/rest/dev/61/5c/20/87/615c2087-2e55-4382-bc29-5e2e346e1112 Service: 70200474251280
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/curation_concerns/base/_attributes.html.erb (814.6ms)
```
Loading GenericFile
```
  Load LDP (171.4ms) http://127.0.0.1:8983/fedora/rest/dev/r7/81/wg/00/r781wg00k Service: 70200474251280
  Search Load (0.2ms)  SELECT  "searches".* FROM "searches" WHERE "searches"."id" IN (2, 1) AND "searches"."id" = ?  ORDER BY updated_at desc LIMIT 1  [["id", 2]]
[CANCAN] Checking read permissions for user: brian@example.com with groups: ["public", "registered"]
[CANCAN] edit_groups: []
[CANCAN] read_groups: ["registered"]
```
Loading member proxies:
```
  Load LDP (135.0ms) http://127.0.0.1:8983/fedora/rest/dev/g7/32/d8/96/g732d8963/members/511d3695-08e2-41d1-bc13-b28e5acadbe9 Service: 70200474251280
  CACHE (0.0ms) http://127.0.0.1:8983/fedora/rest/dev/g7/32/d8/96/g732d8963 Service: 70200474251280
  CACHE (0.0ms) http://127.0.0.1:8983/fedora/rest/dev/g7/32/d8/96/g732d8963/members/511d3695-08e2-41d1-bc13-b28e5acadbe9 Service: 70200474251280
  CACHE (0.1ms) http://127.0.0.1:8983/fedora/rest/dev/g7/32/d8/96/g732d8963 Service: 70200474251280
```
Loading Permissions on the GenericFile. Read, registered
```
  Load LDP (120.1ms) http://127.0.0.1:8983/fedora/rest/dev/77/f5/94/d5/77f594d5-37da-4c5a-af27-37bbda659da8 Service: 70200474251280
```
Loading Permissions on the GenericFile. Write, brian@example.com
```
  Load LDP (119.1ms) http://127.0.0.1:8983/fedora/rest/dev/a0/77/bf/f3/a077bff3-cb1a-4ae9-bb3e-c0b5c44d983e Service: 70200474251280
```
[CANCAN] Checking edit permissions for user: brian@example.com with groups: ["public", "registered"]
[CANCAN] edit_groups: []
[CANCAN] edit_users: ["brian@example.com"]
[CANCAN] decision: true
[CANCAN] Checking edit permissions for user: brian@example.com with groups: ["public", "registered"]
[CANCAN] edit_groups: []
[CANCAN] edit_users: ["brian@example.com"]
[CANCAN] decision: true
[CANCAN] Checking read permissions for user: brian@example.com with groups: ["public", "registered"]
[CANCAN] edit_groups: []
[CANCAN] read_groups: ["registered"]
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/curation_concerns/generic_files/_actions.html.erb (14.6ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/curation_concerns/generic_files/_generic_file.html.erb (2075.8ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/curation_concerns/base/_related_files.html.erb (2772.5ms)
[CANCAN] Checking edit permissions for user: brian@example.com with groups: ["public", "registered"]
[CANCAN] edit_groups: []
[CANCAN] edit_users: ["brian@example.com"]
[CANCAN] decision: true
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/curation_concerns/base/show.html.erb within layouts/curation_concerns/1_column (3741.8ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/shared/_brand_bar.html.erb (0.8ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/shared/_site_search.html.erb (0.6ms)
User is brian@example.com
try is Audio
can is  true
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/shared/_add_content.html.erb (2.5ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/shared/_my_actions.html.erb (4.4ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/shared/_site_actions.html.erb (16.4ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/shared/_title_bar.html.erb (24.9ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/shared/_header.html.erb (41.0ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/shared/_flash_message.html.erb (0.2ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/shared/_footer.html.erb (0.1ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/shared/_ga.html.erb (0.1ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/blacklight-5.14.0/app/views/shared/_ajax_modal.html.erb (0.0ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/layouts/boilerplate.html.erb (409.1ms)
  Rendered /Users/justin/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/curation_concerns-46e3ececfbc8/app/views/layouts/curation_concerns.html.erb (515.2ms)
Completed 200 OK in 4719ms (Views: 4291.7ms | ActiveRecord: 0.5ms)
```

### PR DESCRIPTION
**Pull request by jcoyne**
This reduces the number of API requests to Fedora which speeds things up
a bit. Fixes #159, Ref #198

### CODE REVIEW

**Comment by jcoyne** [comment]
With a simple object with one attached file, this reduced the load time from over 5s to under 1s.

File: app/views/curation_concerns/generic_files/_actions.html.erb [hunk 1]
```diff
@@ -1,16 +1,16 @@
-<% if can?(:edit, generic_file) %>
+<% if can?(:edit, generic_file.id) %>
```
Comment (mjgiarlo) [inline app/views/curation_concerns/generic_files/_actions.html.erb]: Can you say more about this change?

Comment (jcoyne) [inline app/views/curation_concerns/generic_files/_actions.html.erb]: We don't have any abilities defined for `can :edit, GenericFilePresenter`, but we do have an ability for `can :edit, "my-object-id"`

Comment (mjgiarlo) [inline app/views/curation_concerns/generic_files/_actions.html.erb]: Ah, right, that variable holds a presenter. I was fooled.  :cool:

**Comment by mjgiarlo** [comment]
Princetonians want to have a look at this?  @tampakis @jpstroop

**Comment by tampakis** [comment]
:+1:

**Pull request merged**